In [1]:
import pandas as pd
import numpy as np

# Carga de la base de datos con pd.read_excel

In [2]:
df = pd.read_excel("data_sales_final.xlsx")

## Revision de los datos de la base cargada usando: head(), df.tail(), columns, shape, index, info() y isnull().sum().sort_values... se presentan los resultados.

In [3]:
df.head()

id_order purchase_date  year_month  month  day  hour  \
0  CA-2016-152156    2020-11-08      202011     11    7     0   
1  CA-2017-110198    2021-05-01      202105      5    6     0   
2  CA-2017-159793    2021-08-25      202108      8    3     0   
3  CA-2016-117681    2020-04-09      202004      4    4     0   
4  CA-2016-152156    2020-11-08      202011     11    7     0   

        delivery_date   shipping_mode id_client       id_product  ...  \
0 2020-11-11 05:00:00    Second Class  CG-12520  FUR-BO-10001798  ...   
1 2021-05-02 00:00:00     First Class  AG-10900  FUR-BO-10001798  ...   
2 2021-08-29 00:00:00  Standard Class  SV-20365  FUR-BO-10001798  ...   
3 2020-04-14 00:00:00  Standard Class  HF-14995  FUR-BO-10001798  ...   
4 2020-11-11 00:00:00    Second Class  CG-12520  FUR-CH-10000454  ...   

         country       city   category sub_category  \
0  United States  Henderson  Furniture    Bookcases   
1  United States     Tucson  Furniture    Bookcases   
2  United States      Dover  Furniture    Bookcases   
3        Germany     Berlin  Furniture    Bookcases   
4  United States  Henderson  Furniture       Chairs   

                                        product_name production_cost  \
0                  bush somerset collection bookcase      230.098156   
1                  bush somerset collection bookcase      230.098156   
2                  bush somerset collection bookcase      230.098156   
3                  bush somerset collection bookcase      230.098156   
4  hon deluxe fabric upholstered stacking chairs,...      563.102730   

       benefit average_monthly_wage total_sales  diff_purchase_delivery_date  
0    63.723687                 4893     523.920                            3  
1   252.761531                 4893     943.056                            1  
2  -198.236313                 4893     261.960                            4  
3  1632.834218                 3478    2783.325                            5  
4   506.511811                 4893    2195.820                            3  

[5 rows x 24 columns]

In [4]:
df.tail()

id_order purchase_date  year_month  month  day  hour  \
9368  CA-2016-129425    2021-02-17      202102      2    3     0   
9369  CA-2016-129426    2019-03-08      201903      3    5     0   
9370  CA-2016-129427    2021-03-30      202103      3    2     0   
9371  CA-2016-129428    2020-11-25      202011     11    3     0   
9372  CA-2016-129429    2020-05-03      202005      5    7     0   

     delivery_date   shipping_mode id_client       id_product  ...  \
9368    2021-02-19     First Class  CS-12649  OFF-AR-10000937  ...   
9369    2019-03-12  Standard Class  CS-12650  OFF-AR-10002704  ...   
9370    2021-03-31     First Class  CS-12651  OFF-EN-10004206  ...   
9371    2020-11-30  Standard Class  CS-12652  TEC-MA-10004255  ...   
9372    2020-05-05     First Class  CS-12653  TEC-MA-10003589  ...   

          country     city         category sub_category  \
9368  Switzerland    Berna  Office Supplies          Art   
9369  Switzerland    Berna  Office Supplies          Art   
9370  Switzerland  Basilea  Office Supplies    Envelopes   
9371  Switzerland  Basilea       Technology     Machines   
9372  Switzerland  Basilea       Technology     Machines   

                                           product_name production_cost  \
9368  dixon ticonderoga core-lock colored pencils, 4...      144.260686   
9369              boston 1900 electric pencil sharpener        9.911449   
9370                                 multimedia mailers      228.168887   
9371  konica minolta magicolor 1690mf multifunction ...      280.486707   
9372                       cisco 8961 ip phone charcoal      212.190300   

         benefit average_monthly_wage total_sales  diff_purchase_delivery_date  
9368  187.075883                  819    1052.640                            2  
9369    5.068551                  819      14.980                            4  
9370  195.382226                  819     651.720                            1  
9371  315.466342                  819    2559.360                            5  
9372   38.240101                  819     674.811                            2  

[5 rows x 24 columns]

In [5]:
df.columns

Index(['id_order', 'purchase_date', 'year_month', 'month', 'day', 'hour',
       'delivery_date', 'shipping_mode', 'id_client', 'id_product', 'price',
       'quantity', 'client_name', 'segment', 'country', 'city', 'category',
       'sub_category', 'product_name', 'production_cost', 'benefit',
       'average_monthly_wage', 'total_sales', 'diff_purchase_delivery_date'],
      dtype='object')

In [6]:
df.shape

(9373, 24)

In [7]:
df.index

RangeIndex(start=0, stop=9373, step=1)

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9373 entries, 0 to 9372
Data columns (total 24 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   id_order                     9373 non-null   object        
 1   purchase_date                9373 non-null   datetime64[ns]
 2   year_month                   9373 non-null   int64         
 3   month                        9373 non-null   int64         
 4   day                          9373 non-null   int64         
 5   hour                         9373 non-null   int64         
 6   delivery_date                9373 non-null   datetime64[ns]
 7   shipping_mode                9373 non-null   object        
 8   id_client                    9373 non-null   object        
 9   id_product                   9373 non-null   object        
 10  price                        9373 non-null   float64       
 11  quantity                     9373 non-null 

In [9]:
df.isnull().sum().sort_values(ascending=False)

diff_purchase_delivery_date    0
total_sales                    0
purchase_date                  0
year_month                     0
month                          0
day                            0
hour                           0
delivery_date                  0
shipping_mode                  0
id_client                      0
id_product                     0
price                          0
quantity                       0
client_name                    0
segment                        0
country                        0
city                           0
category                       0
sub_category                   0
product_name                   0
production_cost                0
benefit                        0
average_monthly_wage           0
id_order                       0
dtype: int64

### La base contiene 24 columnas y 9373 filas, sin datos vacios. La base se encuentra en condiciones de ser trabajada sin problemas.